server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java

```diff
@@ -92,14 +92,17 @@ public RetentionLease(final String id, final long retainingSequenceNumber, final
         if (timestamp < 0) {
             throw new IllegalArgumentException("retention lease timestamp [" + timestamp + "] out of range");
         }
-        Objects.requireNonNull(source);
         if (source.isEmpty()) {
             throw new IllegalArgumentException("retention lease source can not be empty");
         }
         this.id = id;
         this.retainingSequenceNumber = retainingSequenceNumber;
         this.timestamp = timestamp;
-        this.source = source;
+        this.source = switch (source) {
+            case "ccr" -> "ccr";
+            case ReplicationTracker.PEER_RECOVERY_RETENTION_LEASE_SOURCE -> ReplicationTracker.PEER_RECOVERY_RETENTION_LEASE_SOURCE;
+            default -> source;
+        };
```

Comment:
Do you mind leaving a comment in the code explaining the purpose?
It looks quiet surprising without `StringLiteralDeduplicator` and the explanation in this pr.